crates/miden-objects/src/note/note_tag.rs

```diff
@@ -248,23 +276,54 @@ impl fmt::Display for NoteTag {
 
 impl From<u32> for NoteTag {
     fn from(value: u32) -> Self {
-        Self(value)
+        let prefix = value >> 30;
+        let lower_30_bits = value & 0x3fffffff; // Mask to get the lower 30 bits
+
+        match value & LOCAL_USECASE {
+            NETWORK_ACCOUNT => {
+                // Network, Specific Account (lower 30 bits are the account ID)
+                Self::NetworkAccount(lower_30_bits)
+            },
+            NETWORK_USECASE => {
+                // NetworkUse case (split lower 30 bits into two u16)
+                let use_case = (lower_30_bits >> 16) as u16;
+                let payload = lower_30_bits as u16;
+                Self::NetworkUseCase((NETWORK_USECASE >> 30) as u16 | use_case, payload)
+            },
+            LOCAL_ACCOUNT => {
+                // Local, Any (lower 30 bits are the account ID)
+                Self::LocalAccount(LOCAL_ACCOUNT >> 30 | lower_30_bits)
+            },
+            LOCAL_USECASE => {
+                // Local, Any (split lower 30 bits into two u16)
+                let use_case = (lower_30_bits >> 16) as u16;
+                let payload = lower_30_bits as u16;
+                Self::LocalUseCase((LOCAL_USECASE >> 30) as u16 | use_case, payload)
+            },
+            _ => {
+                // This branch should be unreachable because `prefix` is derived from
+                // the top 2 bits of a u32, which can only be 0, 1, 2, or 3.
+                unreachable!("Invalid prefix encountered: {:b}", prefix);
```

Comment:
No need to compute and include the `prefix` in the message, because we should never encounter it. We can just print the entire `value` here if we want to include it in the message at all.